activerecord/lib/active_record/associations/belongs_to_association.rb

```diff
@@ -148,8 +148,11 @@ def invertible_for?(record)
         end
 
         def stale_state
-          result = owner._read_attribute(reflection.foreign_key) { |n| owner.send(:missing_attribute, n, caller) }
-          result && result.to_s
+          result = Array(reflection.foreign_key).map do |fk|
+            owner._read_attribute(fk) { |n| owner.send(:missing_attribute, n, caller) }
+          end.join(",")
```

Comment:
Do we need to turn it into a string? I think the return value can be an array as all we need is to compare the values so returning an Array should just work 

```suggestion
          end
```

Reply:
@nvasilevski , I tried doing that already. There was a test case failing related to`rails_6_1_associations cache`. What it seems to be doing is ensuring compatibility between Marshal formats. My thought was that we would want to preserve the behaviour unless there are strong reasons not to do so.

Reference: https://buildkite.com/rails/rails/builds/102586#018c4591-20a3-4654-883f-31b465b76125/1095-1105

Reply:
Ah, ok, thanks for clarifying. Since `stale_state` should be internal it's okay to go with a string